### DISCUSSION TITLE
Add an offline mode

### DISCUSSION THREAD

**Discussion by pingshunhuangalex** [monkeytypegame/monkeytype#991] [Ideas]
Call me crazy, but it'll be super awesome if the site somehow have an offline mode (Service Worker maybe) so typing is still possible when I get stuck in an offline environment for a long time, e.g. a plane

**Comment by ObiKeahloa** [comment]
The thing is that most browsers support downloading an entire page to be used offline BUT when you do that on monkeytype it will give you an error saying that the language type is undefined Miodec will have to fix this.

**Reply by Miodec** [reply]
Simply saving the website doesnt work because the languages are stored in different files which are not saved. I dont know how i would go about including those at the moment.

**Reply by ObiKeahloa** [reply]
Is it possible to make it a downloadable file?
As in there could be a download icon on the top right or left of the screen that will download the entire website and all the language files for the language you are using

Edit: When you download the site from a browser it also associates the main .html file with a folder that has all the files required , maybe this can be done but instead the download icon will download not just the entire page but also the language file?

**Reply by Miodec** [reply]
I dont really have experience in doing this, so Im not sure how where i would start with this.

**Reply by ObiKeahloa** [reply]
Well there doesn't seem to be any update about this so I guess it is never getting added?

**Reply by Miodec** [reply]
> Well there doesn't seem to be any update about this so I guess it is never getting added?

It is a possibility but not a priority. I know I say it a lot but I mean, Ill be working on this for probably a long while so every request as a chance.
